### ISSUE TITLE
[URGENT] Bruteforce worker pool implementation does not returns results! 

### ISSUE THREAD

**Issue by Ice3man543** [Ice3man543/subfinder#66]
Hey @Mzack9999,
The worker pool implementation you suggested does not returns result as implemented in bruteforcer. The tasks are run correctly, but it does not returns results in real time. I've taken a look and it seems to be a problem with the worker pool implementation. 

Can you please take a look here. I wasn't able to understand the root cause here. 
PS, do you think we should probably  migrate to original go concurrency model where we have greater control?
Thanks.

**Comment by Mzack9999** [comment]
@Ice3man543 I compared the [latest commit](https://github.com/Ice3man543/subfinder/blob/b501f86eb8ba2f280336892f475323760d66527e/libsubfinder/engines/bruteforce/bruteforce.go) prior to worker pools implementation and can't see big differences. Previously the Bruteforce function couldn't return before all go routines were completed, and so is now. This is the old one:
```
func Bruteforce(state *helper.State) (subdomains []string) {
	wildcard := helper.InitializeWildcardDNS(state)
	if wildcard == true {
		fmt.Printf("\n%s[!]%s Wildcard DNS Detected ! False Positives are likely :-(\n\n", helper.Cyan, helper.Reset)
	}
        
        // Here the functions is stuck until all go routines are completed
	subdomains, err := Process(state.Wordlist, state.Domain, state)
	if err != nil {
		fmt.Printf("\n%v\n", err)
		os.Exit(1)
	}

	return subdomains
}
```
Maybe you want the results printed to terminal as soon as they are discovered in real time and not at the end? In this case I think it's enough to move the print instruction into the consume go routine, something like this in _bruteforce.go_:
```
func consume(args ...interface{}) interface{} {
	target := args[0].(string)
	state := args[1].(*helper.State)
	domain := args[2].(string)
	host := fmt.Sprintf("%s.%s", target, domain)
	ips, err := helper.ResolveHost(host)
	if err != nil {
		return ""
	}

	if len(ips) <= 0 {
		// We didn't found any ips
		return ""
	} else {
		if state.IsWildcard == true {
			result := helper.CheckWildcard(state, ips)
			if result == true {
				// We have a wildcard ip
				return ""
			}
		}
		fmt.Printf("\n[%sBRUTE%s] %s : %s", helper.Info, helper.Reset, host, ips[0])
		return ips[0]
	}
}
```
Let me know if this solves the issue eventually, and I will open a pull request.
I found the actual approach with worker pools more flexible and elegant, since we have just to create the go routine and all the rest is done by the module. But it's just my opinion, do you think we should switch back to classical approach?

**Comment by codingo** [comment]
Closed in #67
